### PR TITLE
Add processing functions

### PR DESCRIPTION
**Pull request by znicholls**
Adds processing functions ``mean_over``, ``median_over``, ``quantile_over`` and ``relative_to_ref_period_mean``

- [x] Implementation finished
- [x] Tests added
- [x] Documentation added
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/openclimatedata/openscm/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

**Comment by znicholls** [comment]
@lewisjared MAGICC related tests failing as per

File: openscm/scmdataframebase.py [hunk 1]
```diff
@@ -827,6 +827,107 @@ def r(*args, **kwargs):
...
+        -------
+        pd.DataFrame
+            The quantiles of the timeseries, grouped by all columns in ``self.meta``
+            other than ``cols``
+        """
+        cols = [cols] if isinstance(cols, str) else cols
+        ts = self.timeseries()
+        group_cols = list(set(ts.index.names) - set(cols))
+
+        return ts.groupby(group_cols).quantile(quantile)
```
Comment (lewisjared) [inline openscm/scmdataframebase.py]: FYI pd.core.groupby.DataFrameGroupBy.quantile also accepts a list of floats to define quantiles

File: tests/integration/test_highlevel.py [hunk 2]
```diff
@@ -470,6 +466,180 @@ def test_timeseries_duplicated(test_scm_df):
     pytest.raises(ValueError, test_scm_df.timeseries, meta=["scenario"])
 
 
+def test_quantile_over_lower(test_processing_scm_df):
+    # not sure how this should work in place, in particular what do you fill
```
Comment (lewisjared) [inline tests/integration/test_highlevel.py]: Easy. It isn't an inplace operation

File: openscm/scmdataframebase.py [hunk 3]
```diff
@@ -827,6 +827,107 @@ def r(*args, **kwargs):
         x._resample_cls = get_resampler(self)
         return x.resample(time=rule, **kwargs)
 
+    def quantile_over(self, cols, quantile):
```
Comment (lewisjared) [inline openscm/scmdataframebase.py]: Could this also take an axis parameter? Would it ever be interesting to see what the nth quantile of a timeseries is?

**Comment by znicholls** [comment]
@lewisjared I'm also going to turn off the MAGICC adapter tests here cause they're not being used anywhere so are just a distraction

File: openscm/scmdataframebase.py (hunk 3 above)
Comment (znicholls) [reply openscm/scmdataframebase.py]: yep e.g. you might want max and min of all the timeseries at each point over say models (so you'd be grouping by scenario)

**Pull request merged**